### PR TITLE
Report cardinality metrics

### PR DESCRIPTION
**Pull request by elgris**
So we know which check goes crazy within 3-days cycle. Based on these data we can communicate to the teams and probably optimize their checks

### CODE REVIEW

**Comment by lmineiro** [comment]
Will the exporter job run on all kairosdb instances? Does it also run on the read stack?

**Review by lmineiro — CHANGES_REQUESTED** [review]
We need to find a way for the exporter job to run only on a single instance. May require coordination, which would be undesired

File: src/main/java/org/kairosdb/core/reporting/CheckCardinalityReporter.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        this.datastore = datastore;
+    }
+
+    @Override
+    public List<DataPointSet> getMetrics(final long now) {
+        final ImmutableList.Builder<DataPointSet> builder = ImmutableList.builder();
+        Iterable<String> metricNames = datastore.getMetricNames();
+        for (String metricName : metricNames) {
+            if (metricName.startsWith(CARDINALITY_PREFIX)) {
```
Comment (mohabusama) [inline src/main/java/org/kairosdb/core/reporting/CheckCardinalityReporter.java:38]: Shouldn't we exclude other non-zmon metrics as well?

File: src/main/java/org/kairosdb/core/reporting/CheckCardinalityReporter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                continue;
+            }
+
+            try {
+                long cardinality = datastore.getWindowCardinality(metricName);
+                final DataPointSet dataPointSet = createDataPointSet(now, metricName, cardinality);
+                builder.add(dataPointSet);
+            } catch (Throwable e) {
+                logger.error("failed reporting cardinality for check '{}': ({}){}", metricName,
```
Comment (mohabusama) [inline src/main/java/org/kairosdb/core/reporting/CheckCardinalityReporter.java]: We assume here a ``check`` which is not KairosDB specific. If it is ok to leak this part, then we could only allow ZMON_METRIC_PREFIX to be included in cardinality counts.

Comment (elgris) [reply src/main/java/org/kairosdb/core/reporting/CheckCardinalityReporter.java]: I will rename it to "metric" to be generic

**Comment by elgris** [comment]
@lmineiro how about exposing "cardinality" with ZMON API and build a check that will periodically get these data from Kairos and report them back as new metrics data?

**Pull request closed without merging**